torch/utils/cpp_extension.py

```diff
@@ -291,16 +291,25 @@ def _get_sycl_arch_list():
 # If arch list returned by _get_sycl_arch_list() is empty, then sycl kernels will be compiled
 # for default spir64 target and avoid device specific compilations entirely. Further, kernels
 # will be JIT compiled at runtime.
+def _get_sycl_target_flags():
+    if _get_sycl_arch_list() != '':
```

Comment:
Why do you build the full arch list just to check if it's empty?

Reply:
User can either specify the empty arch list externally (via `TORCH_XPU_ARCH_LIST` building his extension) or default arch list can be empty if that's how user's pytorch version was originally built (with empty `TORCH_XPU_ARCH_LIST` when building pytorch). Further, empty arch list is still a valid case since it corresponds to JIT runtime compilation. This if handles options difference between JIT and AOT cases.